torch/csrc/jit/mobile/versioned_operators.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <torch/csrc/WindowsTorchApiMacro.h>
+#include <torch/csrc/jit/api/module.h>
+
+#include <cstdint>
+
+namespace torch {
+namespace jit {
+namespace mobile {
+using OperatorFunctor = std::function<void(Stack&)>;
+
+/**
+ * The function to check compatibility of an operator with name. It either
+ * throws error with reason why the operator is not compatible, or returns an
+ * operator functor that is compatible to this runtime.
+ *
+ * @param opname The operator name
+ * @param op_version The operator version in the model file
+ * @param model_version The bytecode version in the model file
+ * @return The operator function pointer that is compatible to this runtime
+ */
+TORCH_API OperatorFunctor operator_resolver(
+    const c10::OperatorName& opname,
+    int64_t op_version,
+    int64_t model_version);
+
+/**
+ * The function to query the supported operators and the supported versions in
+ * current runtime.
+ * @return a map with key: operator name, val: a pair of supported version range
+ * [min, max]
+ */
+TORCH_API std::unordered_map<std::string, std::unordered_set<int64_t>>
```

Comment:
I was using unordered_set here for two reasons. One is that it may be easier for version search for users (like deployment time check). Second I'm not sure if there are situations where "version holes" as @dhruvbird mentioned. Something like the supported versions are {1,3,4} but we don't support 2.

Reply:
Why `int64_t` instead of `uint64_t`?